src/pages/UserProfile.tsx

```diff
@@ -0,0 +1,186 @@
+import Button from '../components/Button/Button';
+import { UserProfile as UserProfileType } from '../types/components/componentType';
+import './userProfile.scss';
+
+const UserProfile: React.FC = () => {
+  const [userProfile, setUserProfile] = useState<UserProfileType | null>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  
+  useEffect(() => {
+    // Модуль із сервісом
+
+    //  тимчасовi дані:
+    setUserProfile({
+      profilePhoto: 'path/to/default/photo.jpg',
+      username: 'Liliia Virsta',
+      email: 'liliia@example.com',
+      dateOfBirth: '2000-01-01',
+      registrationDate: '2022-01-01',
+      lastLoginTimestamp: '2022-02-01',
+    });
+  }, []);
+
+  const formik = useFormik({
+    initialValues: {
+      profilePhoto: userProfile?.profilePhoto || '',
+      username: userProfile?.username || '',
+      email: userProfile?.email || '',
+      dateOfBirth: userProfile?.dateOfBirth || '',
+    },
+    onSubmit: (values) => {
+      // Оновлення профілю в сервісі
+      setUserProfile((prevProfile) => ({
+        ...prevProfile!,
+        profilePhoto: values.profilePhoto,
+        username: values.username,
+        email: values.email,
+        dateOfBirth: values.dateOfBirth,
+      }));
+      setIsEditing(false);
+    },
+    validate: (values) => {
+      const errors: { [key: string]: string } = {};
+
+      if (!values.username) {
+        errors.username = 'Required';
+      }
+      if (!values.email) {
+        errors.email = 'Required';
+      }
+      if (!values.dateOfBirth) {
+        errors.dateOfBirth = 'Required';
+      }
+      return errors;
+    }, 
+  });
+
+  const handleEditClick = () => {
+    setIsEditing((prevIsEditing) => !prevIsEditing);
+  };
+
+  const handleLogout = () => {
+    // Виклик сервісу для виходу (з видаленням токенів)
+    localStorage.removeItem('access_token');
+    localStorage.removeItem('refresh_token');
+    setUserProfile(null);
+    setIsEditing(false);
+  };
+
+  const handleAddPhoto = () => {
+    const fileInput = document.createElement('input');
+    fileInput.type = 'file';
+    fileInput.accept = 'image/png, image/jpeg';
+    fileInput.addEventListener('change', (e: Event) => handlePhotoChange(e));
+    fileInput.click();
+  };
+
+  const handlePhotoChange = (e: Event) => {
+    if (
+      e.target &&
+      e.target instanceof HTMLInputElement &&
+      e.target.files &&
+      e.target.files[0]
+    ) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        const result = (event.target as FileReader)?.result as string;
+        formik.setFieldValue('profilePhoto', result);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+    }
+  };
+
+  
+
+  return (
+    <div className="container-profile">
+      <form className="form" onSubmit={formik.handleSubmit}>
+        {userProfile ? (
+          <div className="profile-width">
+            <h1 className="heading--h1">Personal information</h1>
```

Comment:
Consider organizing your CSS classes and styles more consistently. For example, you have a mix of class names like profile-width and heading--h1

Reply:
Of course. I will change the class names when the front-end team and I agree to use the same type of class names